fast_llm/layers/ssm/csrc/selective_scan/reverse_scan.cuh

```diff
@@ -0,0 +1,401 @@
+/******************************************************************************
```

Comment:
Where is this code taken from? Can we use it as a third party package or submodule instead of copying the whole thing?